memory/src/main/java/org/opensearch/ml/memory/action/conversation/UpdateInteractionRequest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.memory.action.conversation;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+import static org.opensearch.ml.common.conversation.ConversationalIndexConstants.INTERACTIONS_ADDITIONAL_INFO_FIELD;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.core.common.io.stream.InputStreamStreamInput;
+import org.opensearch.core.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentParser;
+
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+public class UpdateInteractionRequest extends ActionRequest {
+    private String interactionId;
+    private Map<String, Object> updateContent;
+
+    private static final Set<String> allowedList = new HashSet<>(Arrays.asList(INTERACTIONS_ADDITIONAL_INFO_FIELD));
+
+    @Builder
+    public UpdateInteractionRequest(String interactionId, Map<String, Object> updateContent) {
+        this.interactionId = interactionId;
+        this.updateContent = filterMapContent(updateContent);
+    }
+
+    public UpdateInteractionRequest(StreamInput in) throws IOException {
+        super(in);
+        this.interactionId = in.readString();
+        this.updateContent = filterMapContent(in.readMap());
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(this.interactionId);
+        out.writeMap(this.getUpdateContent());
```

Comment:
what if updateContent is null?

Reply:
I added a check in the constructor in the filterUpdateContent(), which makes the updateContent an empty map if it's null. Also a null value check is added in the validate() function to throw an exception for the UpdateInteractionRequest.